c7n/resources/cw.py

```diff
@@ -315,3 +316,74 @@ def process_resource_set(self, client, accounts, resources):
             if found:
                 results.append(r)
         return results
+
+
+@LogGroup.action_registry.register('set-encryption')
+class EncryptLogGroup(BaseAction):
+    """Encrypt/Decrypt a log group
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: encrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: absent
+            actions:
+              - type: set-encryption
+                kms_key: kms:key:arn
+                state: True
+
+          - name: decrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: kms:key:arn
+            actions:
+              - type: set-encryption
+                state: False
+    """
+    schema = type_schema(
+        'set-encryption',
+        kms_key={'type': 'string'},
+        state={'type': 'boolean'})
+    permissions = ('logs:AssociateKmsKey', 'logs:DisassociateKmsKey')
+
+    def validate(self):
+        self.key = self.data.get('kms_key')
+
+        if not self.data.get('state', True):
+            return self
+
+        if not self.key:
+            raise ValueError('Must specify either a KMS key ARN or Alias')
```

Comment:
can you add a bit at the end about specifying key with log group, i.e. `Must specify either a KMS key ARN or Alias when setting encryption on log group`

Reply:
Sounds reasonable.